keyring/backends/Windows.py

```diff
@@ -164,7 +164,8 @@ def get_credential(self, service, username):
             res = self._get_password(self._compound_name(username, service))
         # get any first password under the service name
         if not res:
-            res = self._get_password(service)
-            if not res:
-                return None
+            cred = self._get_password(service)
+            res = cred if username is None or username == cred["UserName"] else None
+        if not res:
+            return None
```

Comment:
I noticed these two lines were moved to the top level, but they are only relevant in this branch. Probably it makes sense not to change this behavior.

```suggestion
            if not res:
                return None
```